sequencer/src/header.rs

```diff
@@ -218,18 +218,19 @@ impl Header {
             builder_signature: None,
         };
 
-        // Sign Header with builder wallet from state
-        let signing_key: &SigningKey = builder_address.signer();
-        let header_signature: ecdsa::Signature = signing_key.sign(header.commit().as_ref());
-        // In order to use `types::Signature` for verification we do a
-        // `TryFrom` conversion on the signature. But first we need to
-        // append Etherium's `Electurm` value (can be 27 or 28). There
-        // must be a better way to do this.
-        let header_signature = [header_signature.to_vec(), vec![27]].concat();
+        // Sign our header using its `Committment` as a prehash.
+        let builder_signature = builder_address
+            .sign_hash(types::H256(header.commit().into()))
+            .unwrap();
+
+        assert_eq!(
+            builder_address.address(),
+            parent.fee_info.account().address()
+        );
```

Comment:
What is this assert? It's fine for the parent and the current header to be built/signed by different builders